rs/tests/networking/read_state_test.rs

```diff
@@ -342,6 +373,85 @@ fn test_metadata(env: TestEnv) {
     }
 }
 
+fn test_canister_path(env: TestEnv) {
+    let identities = [
+        PrincipalId::from(get_identity().sender().unwrap()),
+        PrincipalId::from(random_ed25519_identity().sender().unwrap()),
+    ];
+
+    let node = get_first_app_node(&env);
+    let runtime = runtime_from_url(node.get_public_url(), node.effective_canister_id());
+
+    // Create an empty canister
+    let empty_canister: Canister<'_> =
+        block_on(runtime.create_canister_max_cycles_with_retries()).unwrap();
+
+    // Create a canister with some installed WASM
+    let mut installed_canister: Canister<'_> =
+        block_on(runtime.create_canister_max_cycles_with_retries()).unwrap();
+    let wasm = wasm_with_custom_sections(vec![]);
+    block_on(wasm.install_with_retries_onto_canister(&mut installed_canister, None, None)).unwrap();
+
+    // for 2, 1 and 0 controllers...
+    for i in (0..=2).rev() {
```

Comment:
nit: let's be explicit maybe?

```suggestion
    for i in [2, 1, 0] {
```

Reply:
Done 👍